Tests/SwiftIDEUtilsTest/ClassificationTests.swift

```diff
@@ -121,4 +121,379 @@ public class ClassificationTests: XCTestCase {
+        ClassificationSpec(source: "FunnyPrecedence", kind: .identifier),
+        ClassificationSpec(source: "associativity", kind: .keyword),
+        ClassificationSpec(source: "left", kind: .identifier),
+        ClassificationSpec(source: "higherThan", kind: .keyword),
+        ClassificationSpec(source: "MultiplicationPrecedence", kind: .identifier)
+      ]
+    )
+  }
+  
+  public func testInheritence() {
+    assertClassification(
+      "protocol Prot2 : Prot {}", expected: [
+        ClassificationSpec(source: "protocol", kind: .keyword),
+        ClassificationSpec(source: "Prot2", kind: .identifier),
+        ClassificationSpec(source: "Prot", kind: .typeIdentifier)
+      ]
+    )
+  }
+  
+  public func testGenericType() throws {
+    XCTExpectFailure(" '_' is not classified")
+    assertClassification(
+      "func genFn<T : Prot>(_: T) -> Int where T.Blarg : Prot2 {}"
+      , expected: [
+        ClassificationSpec(source: "func", kind: .keyword),
+        ClassificationSpec(source: "genFn", kind: .identifier),
+        ClassificationSpec(source: "T", kind: .identifier),
+        ClassificationSpec(source: "Prot", kind: .typeIdentifier),
+        ClassificationSpec(source: "_", kind: .keyword),
+        ClassificationSpec(source: "T", kind: .typeIdentifier),
+        ClassificationSpec(source: "Int", kind: .typeIdentifier),
+        ClassificationSpec(source: "where", kind: .keyword),
+        ClassificationSpec(source: "T", kind: .typeIdentifier),
+        ClassificationSpec(source: "Blarg", kind: .typeIdentifier),
+        ClassificationSpec(source: "Prot2", kind: .typeIdentifier)
+      ])
+  }
+  
+  public func testStringInterpolationAnchor() {
+    assertClassification(
+      "\"This is string \\(1) interpolation\""
+      , expected: [
+        ClassificationSpec(source: "\"This is string ", kind: .stringLiteral),
+        ClassificationSpec(source: "(", kind: .stringInterpolationAnchor),
+        ClassificationSpec(source: "1", kind: .integerLiteral),
+        ClassificationSpec(source: ")", kind: .stringInterpolationAnchor),
+        ClassificationSpec(source: " interpolation\"", kind: .stringLiteral)
+      ]
+    )
+  }
+  
+  public func testPlaceHolder() {
+    assertClassification(
+      "func <#test1#> () {}"
+      , expected: [
+        ClassificationSpec(source: "func", kind: .keyword),
+        ClassificationSpec(source: "<#test1#>", kind: .editorPlaceholder)
+      ]
+    )
+  }
+  
+  public func testUnownedAndUnsafe() {
+    assertClassification(
+      "unowned(unsafe) var uu"
+      , expected: [
+        ClassificationSpec(source: "unowned", kind: .keyword),
+        ClassificationSpec(source: "unsafe", kind: .identifier),
+        ClassificationSpec(source: "var", kind: .keyword),
+        ClassificationSpec(source: "uu", kind: .identifier)
+      ]
+    )
+  }
+  
+  public func testClosure() {
+    XCTExpectFailure("There is a FIXME in lit-test, not quite sure why")
```

Comment:
I think we weren’t handling closure capture lists properly for a long time (but that’s years in the past). What classification does it produce now?

Reply:
It just produces normal `SyntaxClassification` i.e. `weak` as a keyword, `x` as an identifier, etc..

Reply:
That sounds good to me 👍🏽